common/lib/xmodule/xmodule/modulestore/xml_exporter.py

```diff
@@ -122,72 +192,54 @@ def export_to_xml(modulestore, contentstore, course_key, root_dir, course_dir):
             policy = {'course/' + course.location.name: own_metadata(course)}
             course_policy.write(dumps(policy, cls=EdxJSONEncoder, sort_keys=True, indent=4))
 
-        #### DRAFTS ####
         # xml backed courses don't support drafts!
         if course.runtime.modulestore.get_modulestore_type() != ModuleStoreEnum.Type.xml:
-            # NOTE: we need to explicitly implement the logic for setting the vertical's parent
-            # and index here since the XML modulestore cannot load draft modules
-            with modulestore.branch_setting(ModuleStoreEnum.Branch.draft_preferred, course_key):
-                draft_modules = modulestore.get_items(
-                    course_key,
-                    qualifiers={'category': {'$nin': DIRECT_ONLY_CATEGORIES}},
-                    revision=ModuleStoreEnum.RevisionOption.draft_only
-                )
-
-                if draft_modules:
-                    draft_course_dir = export_fs.makeopendir(DRAFT_DIR)
+            _export_drafts(modulestore, course_key, export_fs, xml_centric_course_key)
 
-                    # accumulate tuples of draft_modules and their parents in
-                    # this list:
-                    draft_node_list = []
 
-                    for draft_module in draft_modules:
-                        parent_loc = modulestore.get_parent_location(
-                            draft_module.location,
-                            revision=ModuleStoreEnum.RevisionOption.draft_preferred
-                        )
-
-                        # if module has no parent, set its parent_url to `None`
-                        parent_url = None
-                        if parent_loc is not None:
-                            parent_url = parent_loc.to_deprecated_string()
-
-                        draft_node = draft_node_constructor(
-                            draft_module,
-                            location=draft_module.location,
-                            url=draft_module.location.to_deprecated_string(),
-                            parent_location=parent_loc,
-                            parent_url=parent_url,
-                        )
-
-                        draft_node_list.append(draft_node)
+def export_library_to_xml(modulestore, contentstore, library_key, root_dir, library_dir):
```

Comment:
@cpennington Libraries don't have things like drafts and they are flat structures, so a lot of the traversal can be side-stepped. The only thing that resembles 'versioning' within the realm of Libraries is the Library Content Module, which keeps reference to its own copy of XBlock settings, to be updated as needed. But any change to the library itself is always live and immediate. Content Modules can then be updated to take advantage of these changes, but will display the old data until they are.